src/web/app/pages-logs/logs-page.component.ts

```diff
@@ -129,11 +141,50 @@ export class LogsPageComponent implements OnInit {
               (e: ErrorMessageOutput) => this.statusMessageService.showErrorToast(e.error.message));
   }
 
+  searchForLogsHistogramView(localDateTime: Observable<number>[]): void {
+    forkJoin(localDateTime)
+      .pipe(
+        concatMap(([timestampFrom, timestampUntil]: number[]) => {
+          this.queryParams = {
+            searchFrom: timestampFrom.toString(),
+            searchUntil: timestampUntil.toString(),
+            severity: 'ERROR',
+          };
+          return this.logService.searchLogs(this.queryParams);
+        }))
+      .pipe(
+        expand((logs: GeneralLogs) => {
+          if (logs.nextPageToken !== undefined) {
+            this.queryParams.nextPageToken = logs.nextPageToken;
+            return this.logService.searchLogs(this.queryParams);
```

Comment:
Be careful, this may end up in infinite loop. In practice it should be fine since ERROR logs are rare, but do take note of it.

Reply:
How should I prevent infinite loop? One way I can think of is to set a maximum page limit (e.g. only retrieve 20 pages), but this will mean it is possible that some error logs from the selected time period are not retrieved.

Reply:
Setting max page limit is fine. If there are still more error logs after e.g. 20 pages, chances are we have much bigger problem to worry about than just being unable to display them.